pkg/output/table.go

```diff
@@ -145,3 +71,70 @@ func ToTable(resources []*schema.Resource) ([]byte, error) {
 	bufw.Flush()
 	return buf.Bytes(), nil
 }
+
+func buildSubResourceRows(t *tablewriter.Table, subresources []*schema.Resource, prefix string) {
+	color := []tablewriter.Colors{
+		{tablewriter.FgHiBlackColor},
+	}
+	if config.Config.NoColor {
+		color = nil
+	}
+
+	for i, r := range subresources {
+		labelPrefix := prefix + "├─"
+		nextPrefix := prefix + "│  "
+		if i == len(subresources)-1 {
+			labelPrefix = prefix + "└─"
+			nextPrefix = prefix + "   "
+		}
+
+		t.Rich([]string{fmt.Sprintf("%s %s", labelPrefix, r.Name)}, color)
+
+		buildCostComponentRows(t, r.CostComponents, nextPrefix, len(r.SubResources) > 0)
+		buildSubResourceRows(t, r.SubResources, nextPrefix)
+	}
+}
+
+func buildCostComponentRows(t *tablewriter.Table, costComponents []*schema.CostComponent, prefix string, hasSubResources bool) {
+	color := []tablewriter.Colors{
+		{tablewriter.FgHiBlackColor},
+		{tablewriter.FgHiBlackColor},
+		{tablewriter.FgHiBlackColor},
+		{tablewriter.FgHiBlackColor},
+		{tablewriter.FgHiBlackColor},
+		{tablewriter.FgHiBlackColor},
+	}
+	if config.Config.NoColor {
+		color = nil
+	}
+
+	for i, c := range costComponents {
+		labelPrefix := prefix + "├─"
+		if !hasSubResources && i == len(costComponents)-1 {
+			labelPrefix = prefix + "└─"
+		}
+
+		t.Rich([]string{
+			fmt.Sprintf("%s %s", labelPrefix, c.Name),
+			formatQuantity(*c.MonthlyQuantity),
+			c.Unit,
+			formatCost(c.Price()),
+			formatCost(c.HourlyCost()),
+			formatCost(c.MonthlyCost()),
+		}, color)
+	}
+}
+
+func formatCost(d decimal.Decimal) string {
+	f, _ := d.Float64()
+	if f < 0.00005 && f != 0 {
+		return fmt.Sprintf("%.g", f)
+	}
+
+	return fmt.Sprintf("%.4f", f)
+}
+
+func formatQuantity(quantity decimal.Decimal) string {
```

Comment:
what do you think about formatting quantities with thousand separators for the table format?

So
```
  aws_lambda_function.lambda
  ├─ Duration                              15,000,000  GB-seconds    2e-05       0.3425      250.0005
  └─ Requests                             100,000,000  requests      2e-07       0.0274       20.0000
  Total                                                                          0.3699      270.0005
```

I suppose we should also do that for hourly/monthly costs?

JSON output should probably not have commas to make it machine-digestible

Reply:
👍 agree, will treat as a separate PR.